include/render.h

```diff
@@ -108,4 +111,8 @@ bool RENDER_StartUpdate(void);
 void RENDER_EndUpdate(bool abort);
 void RENDER_SetPal(uint8_t entry,uint8_t red,uint8_t green,uint8_t blue);
 
+#if C_OPENGL
+extern const std::vector<std::string> builtin_shader_names;
+#endif
+
```

Comment:
DOSBox's source is architected into role-based files, where each `[section]` of the conf file maps 1:1 to a source file. (in addition to more source files to support those roles).

In this case , the `render.cpp` file has the job of rendering pixels from DOS video space up to SDL space, but it's also responsible for managing the `[render]` conf content.

The functional architecture follows the same approach: these role-based files only export the minimal amount of functionality as needed by others.  In this case, the headers provides those API hooks. Above, we see that **Render** shares a couple functions: RENDER_EndUpdate, RENDER_SetPal, and some etc.

Although `Render` does share a couple data structures that are used in the VGA code (Pallete, etc), in general we try to avoid linkages between data structures/containers, and prefer (which makes it harder to refactor). 

So in this case, I would avoid making `sdlmain.cpp` and `render.cpp` both depending on a global vector of strings.  Render is responsible for the shaders, and ideally sdlmain should be as clueless as possible about those details and what structures they're held in, etc.

Instead of sharing this global vector, let's share a function that answers specifically what we want to know - that is, if a shader performs it's own gamma correction or if it expects the linear space.

``` c++
bool RENDER_ShaderManagesOwnGamma(const std::string& shader);

```

(or something like that).

This also avoids the need for `<vector>` to be added to this header.

Reply:
Yeah that would be a better approach, agreed. I'm used to sharing constants or lists of constants freely between modules, but this goes a bit beyond that; it actually shares some internal map, which is not ideal.

Anyway, we won't need this with the pragma approach.